cou/commands.py

```diff
@@ -0,0 +1,343 @@
+        help="Specify machine hostnames(s) to upgrade. This option accepts a single machine "
+        "hostname as well as a stringified comma-separated list for multiple machines. This "
+        "option can be used repetitively.",
+        dest="hostnames",
+        type=str,
+    )
+    dp_upgrade_group.add_argument(
+        "--availability-zone",
+        "--az",
+        action="append",
+        help="Specify availability zone(s) to upgrade. This option accepts a single availability "
+        "zone as well as a stringified comma-separated list for multiple AZs. This option can be "
+        "used repetitively.",
+        dest="availability_zones",
+        type=str,
+    )
+    return dp_subparser
+
+
+def create_plan_subparser(
+    subparsers: argparse._SubParsersAction,
+    subcommand_common_opts_parser: argparse.ArgumentParser,
+    dp_parser: argparse.ArgumentParser,
+) -> None:
+    """Create and configure 'plan' subcommand parser.
+
+    :param subparsers: subparsers that plan subparser belongs to
+    :type subparsers: argparse.ArgumentParser
+    :param subcommand_common_opts_parser parser groups options commonly shared by subcommands
+    :type subcommand_common_opts_parser: argparse.ArgumentParser
+    :param dp_parser: a parser groups options specific to data-plane
+    :type dp_parser: argparse.ArgumentParser
+    """
+    # Arg parser for "cou plan" sub-command
+    plan_parser = subparsers.add_parser(
+        "plan",
+        description="Show the steps COU will take to upgrade the cloud to the next release. "
+        "If upgrade-group is unspecified, plan upgrade for the entire cloud",
+        help="Show the steps COU will take to upgrade the cloud to the next release.",
+        usage="cou plan [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+
+    # Create control-plane and data-plane child commands to plan partial upgrades
+    plan_subparser = plan_parser.add_subparsers(
+        title="Upgrade groups",
+        dest="upgrade-group",
+        help="For more information about a upgrade group, run 'cou plan <upgrade-group>' -h.",
+    )
+    plan_subparser.add_parser(
+        "control-plane",
+        description="Show the steps for upgrading the control-plane components.",
+        help="Show the steps for upgrading the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+    plan_subparser.add_parser(
+        "data-plane",
+        description="Show the steps for upgrading the data-plane components. This is possible "
+        "only if control-plane has been fully upgrade. Otherwise an error will be thrown.",
+        help="Show the steps for upgrading the data-plane components. This is possible "
+        "only if control-plane has been fully upgrade. Otherwise an error will be thrown.",
```

Comment:
```suggestion
        description="Show the steps for upgrading the data-plane components. This is possible "
        "only if control-plane has been fully upgraded, otherwise an error will be thrown.",
        help="Show the steps for upgrading the data-plane components. This is possible "
        "only if control-plane has been fully upgraded, otherwise an error will be thrown.",
```